modules/nodes-vsphere-machine-set-scaling-static-ip.adoc

```diff
@@ -0,0 +1,190 @@
+
+The example in the procedure demonstrates the use of controllers for scaling machines in a machine set.
+
+.Prerequisites
+
+* You included `featureSet:TechPreviewNoUpgrade` as the initial entry in the `install-config.yaml` file.
+* You deployed a cluster that runs at least one node with a configured static IP address.
+
+.Procedure
+. Configure a machine set by specifying IP pool information in the `network.devices.addressesFromPools` schema of the machine set's YAML file: 
++
+[source,yaml]
+----
+apiVersion: machine.openshift.io/v1beta1
+kind: MachineSet
+metadata:
+  annotations:
+    machine.openshift.io/memoryMb: "8192"
+    machine.openshift.io/vCPU: "4"
+  labels:
+    machine.openshift.io/cluster-api-cluster: <infrastructure_id>
+  name: <infrastructure_id>-<role>
+  namespace: openshift-machine-api
+spec:
+  replicas: 0
+  selector:
+    matchLabels:
+      machine.openshift.io/cluster-api-cluster: <infrastructure_id>
+      machine.openshift.io/cluster-api-machineset: <infrastructure_id>-<role>
+  template:
+    metadata:
+      labels:
+        ipam: "true"
+        machine.openshift.io/cluster-api-cluster: <infrastructure_id>
+        machine.openshift.io/cluster-api-machine-role: worker
+        machine.openshift.io/cluster-api-machine-type: worker
+        machine.openshift.io/cluster-api-machineset: <infrastructure_id>-<role>
+    spec:
+      lifecycleHooks: {}
+      metadata: {}
+      providerSpec:
+        value:
+          apiVersion: machine.openshift.io/v1beta1
+          credentialsSecret:
+            name: vsphere-cloud-credentials
+          diskGiB: 120
+          kind: VSphereMachineProviderSpec
+          memoryMiB: 8192
+          metadata: {}
+          network:
+            devices:
+            - addressesFromPools: <1>
+              - group: ipamcontroller.openshift.io
+                name: static-ci-pool
+                resource: IPPool
+              nameservers:
+              - "192.168.204.1" <2>
+              networkName: qe-segment-204
+          numCPUs: 4
+          numCoresPerSocket: 2
+          snapshot: ""
+          template: rvanderp4-dev-9n5wg-rhcos-generated-region-generated-zone
+          userDataSecret:
+            name: worker-user-data
+          workspace:
+            datacenter: IBMCdatacenter
+            datastore: /IBMCdatacenter/datastore/vsanDatastore
+            folder: /IBMCdatacenter/vm/rvanderp4-dev-9n5wg
+            resourcePool: /IBMCdatacenter/host/IBMCcluster//Resources
+            server: vcenter.ibmc.devcluster.openshift.com
+----
+<1> Specifies an IP pool, which lists a static IP address or a range of static IP addresses. The IP Pool can either be a reference to a custom resource definition (CRD) or a resource supported by the `IPAddressClaims` resource handler. The machine controller accesses static IP addresses listed in the machine set's configuration and then allocates each address to each machine. 
```

Comment:
I have a question here. 
`The IP Pool can either be a reference to a custom resource definition (CRD) or a resource supported by the `IPAddressClaims` resource handler.`
I see we mentioned `ipamcontroller.openshift.io` in this doc PR, and looks like it's defined in custom resource definition (CRD) [here](custom resource definition (CRD)) as an sample, so should we also add it as an IPPool sample to doc? 
Or shall we provide a example link of resource supported by the `IPAddressClaims` resource handler?

@rvanderp3 Could you give some help for it? Thanks.

Reply:
Thanks, @rvanderp3 and @sgaoshang . I swapped out three instances of `ipamcontroller.openshift.io` and replaced them with `ipamcontroller.example.io`.

Is the following description still OK?

> Specifies an IP pool, which lists a static IP address or a range of static IP addresses. The IP Pool can either be a reference to a custom resource definition (CRD) or a resource supported by the `IPAddressClaims` resource handler. The machine controller accesses static IP addresses listed in the machine set's configuration and then allocates each address to each machine.

Reply:
Thanks @rvanderp3 , could you pls tell how to set IPPool range in the IPAddressClaim?